frontend/src/app/shared/components/show-changes/show-changes.component.ts

```diff
@@ -0,0 +1,251 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See COPYRIGHT and LICENSE files for more details.
+//++
+
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  HostBinding,
+} from '@angular/core';
+
+import { I18nService } from 'core-app/core/i18n/i18n.service';
+import { UntilDestroyedMixin } from 'core-app/shared/helpers/angular/until-destroyed.mixin';
+import { WorkPackageViewFiltersService } from 'core-app/features/work-packages/routing/wp-view-base/view-services/wp-view-filters.service';
+import { HalResourceService } from 'core-app/features/hal/services/hal-resource.service';
+import SpotDropAlignmentOption from 'core-app/spot/drop-alignment-options';
+import { WeekdayService } from 'core-app/core/days/weekday.service';
+import { DayResourceService } from 'core-app/core/state/days/day.service';
+import { IDay } from 'core-app/core/state/days/day.model';
+import { take } from 'rxjs/operators';
+import { TimezoneService } from 'core-app/core/datetime/timezone.service';
+import { ConfigurationService } from 'core-app/core/config/configuration.service';
+
+@Component({
+  selector: 'op-show-changes',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  templateUrl: './show-changes.component.html',
+  styleUrls: ['./show-changes.component.sass'],
+})
+export class OpShowChangesComponent extends UntilDestroyedMixin implements AfterViewInit {
+  @HostBinding('class.op-show-changes') className = true;
+
+  public opened = false;
+
+  public dropDownDescription = '';
+
+  public nonWorkingDays:IDay[] = [];
+
+  public selectedDate = '';
+
+  public selectedTimezoneFormattedTime = '';
+
+  public filterSelected = false;
+
+  public timeZoneSelected = false;
+
+  public daysNumber = 0;
+
+  public tooltipPosition = SpotDropAlignmentOption.BottomRight;
+
+  public text = {
+    toggle_title: this.I18n.t('js.show_changes.toggle_title'),
+    header_description: this.I18n.t('js.show_changes.header_description'),
+    clear: this.I18n.t('js.show_changes.clear'),
+    apply: this.I18n.t('js.show_changes.apply'),
+    show_changes_since: this.I18n.t('js.show_changes.show_changes_since'),
+    time: this.I18n.t('js.show_changes.time'),
+    help_description: this.I18n.t('js.show_changes.help_description'),
+    timeZone: this.configuration.isTimezoneSet() ? moment().tz(this.configuration.timezone()).zoneAbbr() : 'local',
+    time_description: () => this.I18n.t('js.show_changes.time_description', { time: this.selectedTimezoneFormattedTime, days: this.daysNumber }),
+  };
+
+  public showChangesAvailableValues = [
+    {
+      value: '0',
```

Comment:
I'd prefer we'd use constants instead of integers to map between the different states. That would make the switch below more readable, too